include/uart.h

```diff
@@ -64,6 +65,62 @@ extern "C" {
  */
 #define UART_ERROR_BREAK    (1 << 3)
 
+/**
+ * @brief UART controller configuration structure
+ *
+ * @param baudrate   Baudrate setting in bps.
+ * @param rx_timeout RX timeout in ms.
```

Comment:
Should we stick with timeout in miliseconds, microseconds or use bytes as timeout? Alternatively we can add flag that allows to choose which way will be used.
@nordic-krch @pfalcon @tbursztyka

Reply:
I think we should use SI time units. And unless there're really good reasons to use microseconds (should be presented/referred to), milliseconds should be used, as elsewhere in Zephyr.

For cases where character-space based timeouts are desirable, these can be calculated from baud rate. But then again, what order of timeouts we're talking about? For example, 1char @115200 baud is ~87us. You guys were talking about hardware which supports timeouts - which are examples and how does it work?